### PR TITLE
Add release notes for FreeIPA 4.11.0-beta

### PR DESCRIPTION
**Pull request by antoniotorresm**

### CODE REVIEW

File: src/release-notes/4-11-0.rst [hunk 1]
```diff
@@ -0,0 +1,1869 @@
...
+      FreeIPA can now authenticate users with the help of OAuth 2.0
+      identity providers supporting OAuth 2.0 Device Authorization Flow.
+      IdPs known to work are Keycloak, Microsoft Azure, Google, Github,
+      and Okta. Details on how to use Keycloak can be found in FreeIPA
+      workshop:
+      https://freeipa.readthedocs.io/en/latest/workshop/12-external-idp-support.html
+
+--------------
+
+-  9083: Support MIT Kerberos KDB version 9
```
Comment (abbra) [inline src/release-notes/4-11-0.rst:60]: Can be removed, this is already in 4.10+

File: src/release-notes/4-11-0.rst [hunk 2]
```diff
@@ -0,0 +1,1869 @@
...
+
+      FreeIPA now supports MIT Kerberos 1.20. Resource-based constrained
+      delegation is not yet implemented.
+
+--------------
+
+-  9150: Remove 'Remove' button from subid page
```
Comment (abbra) [inline src/release-notes/4-11-0.rst:67]: Can be removed, already in 4.9/4.10

File: src/release-notes/4-11-0.rst [hunk 3]
```diff
@@ -0,0 +1,1869 @@
...
+
+      subid ranges cannot be removed. A button in Web UI subid
+      management page to remove the range was removed to not confuse
+      users
+
+--------------
+
+-  9159: [RFE] ipa-client-install should provide option to enable subid:
```
Comment (abbra) [inline src/release-notes/4-11-0.rst:75]: Can be removed, already released in 4.9+

**Review by abbra — COMMENTED** [review]
I checked all and these tickets all can be removed because they already released in the previous versions. The only tickets that should be present are the ones related to the passkeys and certificate pruning (I think?).

**Comment by dav-pascual** [comment]
> FWIW the preview link returns a 404

@rcritten fixed

**Comment by rcritten** [comment]
> > FWIW the preview link returns a 404
> 
> @rcritten fixed

The link title is a bit misleading. I expected it to take me to the rendered page, https://www.freeipa.org/pull/15/release-notes/4-11-0.html . Instead it goes to the new root of the whole site and I had to navigate to the page. That means that reviewers need knowledge of where an updated page will live.

File: src/release-notes/4-11-0.rst [hunk 4]
```diff
@@ -0,0 +1,1732 @@
...
+      in the default store by the installer already.
+
+--------------
+
+-  9273: [RFE] Support IPA CA installation on an HSM
+
+      FreeIPA CA can now be deployed with a hardware security module as
+      a CA storage device. Supported use case details can be found in
+      HSM design document:
+      https://freeipa.readthedocs.io/en/ipa-4-10/designs/hsm.html
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:55]: Please remote. This work is incomplete.

File: src/release-notes/4-11-0.rst [hunk 5]
```diff
@@ -0,0 +1,1732 @@
...
+-  9228: ipa-client-install does not maintain server affinity during
+   installation
+
+      ipa-client-install will use a single server for the duration of
+      the installation process, either one discovered or provided on the
+      command-line. Previously it would use a temporary configuration to
+      do enrollment, then switch to a final one for the remaining
+      operations. This could lead to the installer talking with multiple
+      servers. If the client installer is faster than replication this
+      could lead to errors.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:28]: This is not new in 4.11.0.

File: src/release-notes/4-11-0.rst [hunk 6]
```diff
@@ -0,0 +1,1732 @@
...
+
+--------------
+
+-  9237: Show order in sudo rule list in web interface
+
+      In the 'sudo rules' page, the WebUI is now displaying a 'sudo
+      order' column so that the users can easily see which rules
+      override other rules based on their order.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:36]: This is not new in 4.11.0.

File: src/release-notes/4-11-0.rst [hunk 7]
```diff
@@ -0,0 +1,1732 @@
...
+
+--------------
+
+-  9258: Do not add TLS CA configuration to ldap.conf anymore
+
+      FreeIPA client installer does not add explicit TLS CA
+      configuration to OpenLDAP's ldap.conf anymore. Since OpenLDAP
+      2.4.45, explicit CA configuration is not required as OpenLDAP uses
+      the default CA store provided by OpenSSL and IPA CA is installed
+      in the default store by the installer already.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:46]: This is not new in 4.11.0.

File: src/release-notes/4-11-0.rst [hunk 8]
```diff
@@ -0,0 +1,1732 @@
...
+
+      FreeIPA CA can now be deployed with a hardware security module as
+      a CA storage device. Supported use case details can be found in
+      HSM design document:
+      https://freeipa.readthedocs.io/en/ipa-4-10/designs/hsm.html
+
+--------------
+
+-  9287: [RFE] makeapi should validate the generated API doc vs stored
+   doc
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:60]: This is not new in 4.11.0 (it was introduced in 4.10.2)

File: src/release-notes/4-11-0.rst [hunk 9]
```diff
@@ -0,0 +1,1732 @@
...
+
+--------------
+
+-  9294: Enable the certificate pruning job in PKI
+
+      Removing (pruning) expired certificates is supported when Random
+      Serial Numbers are enabled. One cannot upgrade from sequential
+      serial numbers to random. This feature is enabled using the
+      ipa-acme-manage(1) command.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:69]: This is not new in 4.11.0 (it was introduced in 4.10.2)

File: src/release-notes/4-11-0.rst [hunk 10]
```diff
@@ -0,0 +1,1732 @@
...
+--------------
+
+-  9298: [Tracker] Nightly test failure (updates-testing) in
+   test_acme.py::TestACME::test_certbot_certonly_standalone
+
+      With Certbot update to 2.0.0, Certbot defaults to ECDSA
+      certificate private keys for all new certificates. PKI ACME cert
+      profile supports only rsa private keys, meaning that the key type
+      needs to be forced to rsa when requesting an ACME certificate,
+      using certbot --key-type rsa [...]
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:80]: This is not new in 4.11.0 (it was introduced in 4.10.2)

File: src/release-notes/4-11-0.rst [hunk 11]
```diff
@@ -0,0 +1,1732 @@
...
+
+-  9354: Implement resource-based constrained delegation
+
+      FreeIPA provides initial implementation of resource-based
+      constrained delegation (RBCD) for Kerberos services. RBCD and
+      other Kerberos delegation services described in the design
+      document:
+      https://freeipa.readthedocs.io/en/latest/designs/rbcd.html. The
+      initial implementation works for FreeIPA services, work on
+      supporting cross-realm RBCD continues.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:101]: This is not new in 4.11.0 (it was introduced in 4.10.2)

File: src/release-notes/4-11-0.rst [hunk 12]
```diff
@@ -0,0 +1,1732 @@
...
+      https://pagure.io/freeipa/c/3f1b373cb2028416e40a26e3dd99b0f4c82525c7.
+      In addition, a 'full PAC' signature type was added to MIT Kerberos
+      1.21. FreeIPA will support the new signature when running against
+      newer MIT Kerberos version. For older versions, please see
+      https://pagure.io/freeipa/c/9cd5f49c74f28dbe070b072b394747a039cef463.
+      This new PAC signature will be required by default by Active
+      Directory in July 2023 for S4U requests, and opt-out will no
+      longer be possible after October 2023. We recommend upgrading to
+      newer versions of FreeIPA-based distributions to avoid
+      interoperability break.
```
Comment (rcritten) [inline src/release-notes/4-11-0.rst:121]: This is not new in 4.11.0 (it was introduced in 4.10.2)

**Comment by flo-renaud** [comment]
Here are the tickets fixed between 4.10.2 and 4.11:
- #9263 Add support for passkey authentication type in kdb driver 
- #9262 Add "passkey" authentication type 
- #9261 Add CLI and WebUI to register a passkey for a user 
- #9389 Nightly test failure in test_webui_service
- #9003 ipa-server-install not validating hostname != domain
- #9331Better handling of the command line and web UI cert search and/or list features
- #9396 Renaming user or group with --setattr does not check supported formats
- #9381Race condition in ipa-server-upgrade where pki-tomcat needs dirsrv while it's stopped
- #9385 Upgrade to 4.9.10-6.0.1 fails: attributes are managed by topology plugin
- #9330 Nightly test failure (testing_master_pki): TestBackupReinstallRestoreWithKRA::test_full_backup_reinstall_restore_with_vault
- #9317 Distinguish between different location meaning
- #9404 Nightly test failure in test_integration/test_backup_and_restore.py::TestBackupAndRestoreWithReplica::test_full_backup_and_restore_with_replica 
- #9399 Nightly tests(rawhide): test_epn not compatible with dnf5
- #9402 OTP authentication failure on s390x
- #9378 [RFE] Descriptive error message in ipa user-add
- #9409 freeipa uses ssl.match_hostname() which was removed from Python 3.12
- #9416 Better error description when managing a user with '--idp'
- #9386 Update SELinux policy
- #9336 Allow custom real name in IPA-EPN
- #9419 Nightly test failure in test_epn.py::TestEPN::test_EPN_config_file

The ones worth mentioning in the highlights are #9263, #9262 and #9261 (passkey support), maybe #9336.
The pruning was already added in 4.10.2

**Comment by antoniotorresm** [comment]
Thanks @flo-renaud, I've updated the release notes with your list.

**Comment by abbra** [comment]
This looks better now.

Two things I am concerned about at this point:
- marking clear it is a beta version of 4.11.0 release
- marking clear this release and passkeys are only supported on Fedora Rawhide (F39) due to dependency on SSSD 2.9.0 with passkeys support enabled.

**Comment by rcritten** [comment]
In the past the MAJOR/MINOR would be set to the release name (4.11.0) and we'd use the IPA_VERSION_PRE_RELEASE to set the pre-release version.

And tag it as:  something like beta_3-3-0-0 (for the 3rd beta of 3.0.0).

For example, 6fb802152add24aa1842f4adccf59b23850ab336

File: src/release-notes/4-11-0-beta.rst [hunk 13]
```diff
@@ -0,0 +1,803 @@
...
+
+    - delegating authentication to an external RADIUS server
+
+Initial implementation only supports physical FIDO2 devices and requires use of
+SSSD 2.9.1 or later version, with passkeys support enabled. At the time of
+FreeIPA 4.11.0 release this version is only available in developing versions of
+Fedora (39/Rawhide), Debian (testing/sid), Ubuntu (Mantic) and several other
+distributions.
+
+Two major missing features in passwordless authententication integration in
```
Comment (flo-renaud) [inline src/release-notes/4-11-0-beta.rst:50]: Typo: authen**ten**tication

**Review by flo-renaud — COMMENTED** [review]
Hi @antoniotorresm 
thanks for the PR. I only noted a small typo, please see my inline comment.

File: src/release-notes/4-11-0-beta.rst (hunk 13 above)
Comment (abbra) [reply src/release-notes/4-11-0-beta.rst]: thanks, I fixed this.

**Pull request merged**